src/maxminddb.c

```diff
@@ -375,7 +375,11 @@ LOCAL int map_file(MMDB_s *const mmdb)
     ssize_t size;
     int status = MMDB_SUCCESS;
 
-    int fd = open(mmdb->filename, O_RDONLY);
+    int flags = O_RDONLY;
+#if _POSIX_VERSION >= 200809L
```

Comment:
Rather than comparing against this version, could we just check whether or not the feature we care about (`O_CLOEXEC`) is available?